docs/data-retention-mechanisms.md

```diff
@@ -0,0 +1,65 @@
+# Data retention mechanisms
+
+There are several mechanisms regarding the retention of data in EmailAlertApi.
+This doc gives a brief overview of each.
+
+## Deactivating subscribers
+
+We deactivate subscribers for a couple of different scenarios. One being where
+the user unsubscribes via either following an unsubscribe link in the email
+or via the manage subscriptions page.
+
+Another way a user is deactivated is if their subscription is ended on our side
+because the topic they're signed up for no longer exists. This happens via an
+unpublished rabbitMQ message which is eventually handled by the
+[UnpublishHandleService]. This is only applicable for subscriptions to the
+topic taxonomy.
+
+[UnpublishHandleService]: https://github.com/alphagov/email-alert-api/blob/master/app/services/unpublish_handler_service.rb
+
+## Nullifying deactivated subscribers
+
+We nullify the email addresses of subscribers after they've been deactivated
+for over 28 days. This means that the address field on the subscriber record is
+updated to nil once the threshold has been reached.
+
+We use the [NullifyDeactivatedSubscribersWorker] to do this which runs [every
+hour].
+
+[every hour]: https://github.com/alphagov/email-alert-api/blob/master/config/sidekiq.yml#L26-L28
+[NullifyDeactivatedSubscribersWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/nullify_deactivated_subscribers_worker.rb
+
+## Archiving emails
+
+We archive emails to S3 using the [EmailArchiveWorker]. Emails are eligible for
+archiving immediately after creation as long as they haven't already been
+archived (based on `archived_at` value).
+
+The [archive worker runs every hour].
+
+[EmailArchiveWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[archive worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L29-L31
+
+## Deleting emails
+
+We delete emails after they've been archived for a minimum of 7 days. Emails get
```

Comment:
```suggestion
We delete emails 7 days after they've been archived. Emails get
```

Minor: the "minimum" wording here sounded like there would be other days in the future where we might archive an email.